### PR TITLE
Localizations of affine algebras and affine schemes

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
We introduce localizations of affine algebras (i.e. quotients of polynomial rings by ideals) and their morphisms. This provides the basis for the reworked affine schemes. 

Documentation and tests might still be insufficient, but this will be added within the next days.

### CODE REVIEW

**Review by thofma — COMMENTED** [review]
I am not in a position to judge the mathematics.

Looks good on the technical side, except for the `try catch` things, which in my opinion should be removed.

It is super slow and more importantly opens the door wide for bugs. For example, one would have to check the error, that really the error one expected is thrown. I am also a bit sceptical about the usage. The `@warn` is noisy and not helping. Either we guarantee that the result is correct in which case the warning is of no use and just confusing. Or we cannot guarantee that the result is correct, in which case we should throw an error.

**Comment by thofma** [comment]
There are also a lot of paths/functions which are not called.

**Comment by HechtiDerLachs** [comment]
I was also surprised when I found the try-catch block. It's not even important. I guess, I just wanted to 'try' something new at that point. So it's going to be removed.

And the non-called functions: These will hopefully be covered by the tests for the schemes to come.

**Comment by HechtiDerLachs** [comment]
To me it looks like this can be merged. The failure of the tests is not due to my code, as it seems. I will leave this 'as is' for now and wait for your approval (or complaints).

**Comment by simonbrandhorst** [comment]
As far as I checked the mathematics look good. But for a thorough check the pull request is too long. 
I would recommend to split the next pull request into smaller chunks (if that makes sense). 
Please resolve the merge  conflict.

**Comment by HechtiDerLachs** [comment]
About the mathematics: The foundations for localizations of polynomial rings had already been part of the previous PR. What's essentially new is the part on products of multiplicative sets and the `saturated_ideal`. If anyone found the time to double check this, that would probably be nice. Also, if anyone knew other common references for this besides Wikipedia. I mean: it's not too difficult to check all this yourself and I was quite explicit in the comments in the source code. But people might feel more comfortable if we were providing them with a textbook reference. 

Apart from that, I'm afraid we will just have to wait for people to start using that stuff to pin down any further possible mistakes. 

Why was there a merge conflict in the first place? It doesn't make sense, since there were only two lines added compared to the previous version. Is there a reason this doesn't go through as a simple modification?

**Comment by HechtiDerLachs** [comment]
Question to the reviewers: Is there anything specific that you are waiting for from my side? 

For us, it would be convenient to have this merged soon, so that we have a reliable basis on which we can continue. I understand that this is a rather big PR, but if the code looks OK, then consistency of the mathematics will be tested with upcoming further developments, anyway. If it makes you feel safer, then I could move more parts to the `experimental` section, for instance the `mpolyquo-localizations.jl`. Let me know if you prefer this.

**Comment by thofma** [comment]
In the docs there are lots of cases of
```
``φ : P[T⁻¹] → Q[U⁻¹]``
```
so lots of unicode. Is this supposed to render as LaTeX? Will this render properly?

**Comment by thofma** [comment]
There are also some
```
`saturated ideal`
```
which should probably be `*saturated ideal*` or `**saturated ideal**`?

**Comment by thofma** [comment]
We could also just merge and then work out these formatting issues later (they will be clearly visible in the documentation). What do you prefer?

**Comment by HechtiDerLachs** [comment]
On my machine the documentation renders fine. It seems that some of the unicode is even automatically converted to latex for the html version, probably because of the mathjax. 

About the saturated ideal: grep doesn't give me any instances with the missing underscore. Can you point one out directly?

File: docs/src/CommutativeAlgebra/localizations.md [hunk 1]
```diff
@@ -184,21 +233,65 @@ MPolyLocalizedRingElem{
...
+if ``\mathbb k`` is a Noetherian ring, any localization ``W = R[U^{-1}]`` of a 
+multivariate polynomial ring ``R = \mathbb k[x_1,\dots,x_n]`` is again Noetherian and 
+any ideal ``I \subset W`` is of the form ``I = I'\cdot W`` for some ideal ``I' \subset R``. 
+This correspondence is not 1:1 but for any ideal ``I \subset W`` we always 
+have that 
+```math
+  J = \left\{ x\in R : \exists u \in U : u\cdot x \in I \right\}
+```
+is the unique element which is maximal among all ideals ``I'`` in ``R`` for 
+which ``I = I'\cdot W``. We call this the `saturated ideal` of the localization
```
Comment (thofma) [inline docs/src/CommutativeAlgebra/localizations.md]: Here.

Comment (HechtiDerLachs) [reply docs/src/CommutativeAlgebra/localizations.md]: Ok, thx. I was looking in the *.jl files ony... Yes, this should be corrected.

File: experimental/Schemes/AffineSchemes.jl [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+export ⊂
+
+export is_open_embedding, is_closed_embedding, hypersurface_complement, subscheme, name_of, set_name!
+export closure, product
+
+export SpecMor
+export pullback, domain, codomain, preimage, restrict, graph
+
+@Markdown.doc """
+Scheme{BaseRingType<:Ring, BaseRingElemType<:RingElement}
```
Comment (thofma) [inline experimental/Schemes/AffineSchemes.jl]: Needs 4 spaces for proper rendering.

**Pull request merged**